### ISSUE TITLE
Understanding Text   Spacing

### ISSUE THREAD

**Issue by michael-n-cooper** [w3c/wcag#347]
_From @awkawk on March 2, 2018 15:19_

<p>The <a   href="https://www.w3.org/TR/WCAG21/#text-spacing"   target="_blank">1.4.12 Text Spacing (AA) SC</a> has draft   Understanding content that can be viewed at <a   href="https://rawgit.com/w3c/wcag21/text-spacing/understanding/21/text-spacing.html"   target="_blank">Understanding Text   Spacing</a>.</p><p>Please review the Understanding document   and submit a comment on this issue for any changes that are needed to the   content.</p>

_Copied from original issue: w3c/wcag21#783_

**Comment by michael-n-cooper** [comment]
_From @becka11y on March 27, 2018 20:41_

This understanding document is very good - I appreciate the addition of visual examples.

If I follow the Allow for Spacing Override link under the sufficient techniques in the Understanding document (https://www.w3.org/WAI/GL/wiki/Allowing_for_Spacing_Override). There is a very good explanation of how to test this SC.  The test explicitly requires the screen size set to 1280x768 for testing but I can't find out why that resolution is necessary.  I don't see anything in the SC text that implies a specific screen size?  

FWIW most responsive web sites pass at 1280x768 using the testing technique described.  More fail at lower resolutions (1024x768).  Thus, since the size used for testing does make a difference in the outcome it should be explained.

I realize I am getting ahead of myself in reading and commenting on the sufficient technique but if screen size is important it should be part of the SC.

thanks

**Comment by michael-n-cooper** [comment]
_From @lauracarlson on March 28, 2018 12:5_

Thanks for the comment @becka11y 

Becky wrote:
> The test explicitly requires the screen size set to 1280x768 for testing but I can't find out why that resolution is necessary. I don't see anything in the SC text that implies a specific screen size?
> 
> FWIW most responsive web sites pass at 1280x768 using the testing technique described. More fail at lower resolutions (1024x768). Thus, since the size used for testing does make a difference in the outcome it should be explained.
> 
> I realize I am getting ahead of myself in reading and commenting on the sufficient technique but if screen size is important it should be part of the SC.

@alastc @allanj-uaag Do you recall the rationale for 1024x768? Think that may also be applicable to the Reflow Understanding doc #781 and  the Draft [Allowing for Reflow Technique](https://www.w3.org/WAI/GL/wiki/Allowing_for_Reflow).

**Comment by michael-n-cooper** [comment]
_From @alastc on March 28, 2018 14:38_

It looks like part of the reflow method has been copied over?

The 1280 width comes from there, because 1280px wide at 400% = 320px wide, and Reflow explicitely references that size.

For text-spacing, the width and zoom level are not important. Or rather, not dependant: theoretically text-spacing should work at any size/zoom level.

In practice we have found very few differences at different sizes, if you don't set fixed heights it's generally ok.

For testing spacing, perhaps we could say somehting like: Where a page has mutiple layouts (e.g. in a repsonsive design) text-spacing should be tested in each layout.

**Comment by michael-n-cooper** [comment]
_From @lauracarlson on March 28, 2018 15:30_

Hi @alastc,

[Alastair wrote](https://github.com/w3c/wcag21/issues/783#issuecomment-376910421),

> It looks like part of the reflow method has been copied over?

Ah yes. Now I recall. Last summer we were testing [both spacing and reflow SCs](https://www.w3.org/WAI/GL/wiki/Results_of_Bookmarklet_Tests_for_Issue_78) so the method was combined.

> The 1280 width comes from there, because 1280px wide at 400% = 320px wide, and Reflow explicitely references that size.
>
> For text-spacing, the width and zoom level are not important. Or rather, not dependant: theoretically text-spacing should work at any size/zoom level.
> 
> In practice we have found very few differences at different sizes, if you don't set fixed heights it's generally ok.
> 
> For testing spacing, perhaps we could say somehting like: Where a page has mutiple layouts (e.g. in a repsonsive design) text-spacing should be tested in each layout.

I've updated the Allowing for [Spacing Override Technique](https://www.w3.org/WAI/GL/wiki/Allowing_for_Spacing_Override#Expected_Results) with that verbiage.

Do we need to repeat "Where a page has multiple layouts (e.g. in a responsive design) text spacing should be tested in each layout" in both the Understanding Doc and the Technique? Or is it better not to go into such detail in Understanding?

Thank you!

**Comment by michael-n-cooper** [comment]
_From @iamjolly on March 28, 2018 15:31_

> For testing spacing, perhaps we could say somehting like: Where a page has mutiple layouts (e.g. in a repsonsive design) text-spacing should be tested in each layout.

👍 
That seems very logical @alastc. And thanks @lauracarlson!

**Comment by michael-n-cooper** [comment]
_From @awkawk on March 28, 2018 15:35_

@iamjolly We do have a note under "full pages" in the conformance section that says:
A full page includes each variation of the page that is automatically generated by the page for various screen sizes. Each of these variations needs to conform (or needs to have a conforming alternate version) in order for the entire page to conform.

**Comment by michael-n-cooper** [comment]
_From @iamjolly on March 28, 2018 15:41_

Thanks, @awkawk. I didn't originally notice that note, but it's good to know it is in there and the other info has been updated, too. You *all* are doing fine work on this.

**Comment by michael-n-cooper** [comment]
_From @lseeman on April 22, 2018 17:22_

Can we add 
Ideally, there should be sufficient white space between blocks of text such as sections and call out boxes to make them easy to differentiate for people with cognitive disabilities. 

@eadraffan can you improve this wording?

**Comment by michael-n-cooper** [comment]
_From @alastc on April 22, 2018 21:44_

Given the inclusion of the paragraph spacing bullet in the SC, I think the intent and benefits should include that aspect as well. E.g.

>  increase space between lines, words, **letters and paragraphs** to increase reading speed.

Extending to sections & call out boxes (Lisa's line above) needs to be clearly 'additional' to the SC, I think that could be done at the end of the Intent section.

**Comment by michael-n-cooper** [comment]
_From @eadraffan on April 23, 2018 10:39_

Reading the document on https://rawgit.com/w3c/wcag21/text-spacing/understanding/21/text-spacing.html - I have highlighted these suggestions in bold

The intent of this Success Criterion (SC) is to help ensure that people with disabilities who override spacing **can continue to read the text**. For example, people with low vision or dyslexia may override an author's set spacing to enable readability or increased reading speed. **_delete as this is explained in the next sentence and could confuse 'It allows for a buffer'_** Increased spacing between lines, words, and letters has benefits for people who want buffers between text by override author settings via user stylesheet, bookmarklet, extension, or application. They may also need to change font family (e.g. to a wider one) than the author has set to effectively read text. They may increase spacing between lines, words, and letters to effectively read text. Line spacing also helps with tracking. Tracking is following along lines of text, including getting from the end of one line to the beginning of the next line.

Also I think there are 2 errors in the exception section. Correction indicated by bold
Character Spacing
 Individual characters in words remained **intact** though they were spaced a bit further apart
 
Word Spacing
Words were spaced farther apart. In languages that do not have words (e.g. Japanese) applying **word** spacing had no effect. This is expected.

**Comment by michael-n-cooper** [comment]
_From @lauracarlson on April 23, 2018 14:8_

Hi @eadraffan, 

Thank you very much for reviewing the document. I [fixed the 2 typos](https://github.com/w3c/wcag21/commit/2bd4aa3fcfae182d0325ae84281d0b6bc0553cfe).

I am wondering why we would want to change the first sentence from:

> The intent of this Success Criterion (SC) is to help ensure that people with disabilities who override spacing can read text. 

to

>The intent of this Success Criterion (SC) is to help ensure that people with disabilities who override spacing can continue to read the text.

For some people with low vision, spacing provides the ability to read text not just to continue to read it. Please consult the [Accessibility Requirements for People with Low Vision (3.4 Spacing for Reading)](https://rawgit.com/w3c/wcag21/text-spacing/understanding/21/text-spacing.html):

> Some people need more space between lines to be able to read text.
 
> Some people need more space between letters to read text.

> Some people need more space between words to read text.

Thank again,
Laura

**Comment by michael-n-cooper** [comment]
_From @lauracarlson on April 23, 2018 14:25_

Hi @eadraffan ,

I wonder if we can simplify your proposed sentence:

> Increased spacing between lines, words, and letters has benefits for people who want buffers between text by override author settings via user stylesheet, bookmarklet, extension, or application.

It has an an [average grade level of about 18](https://www.webpagefx.com/tools/read-able/check.php). It should be understood by 23 to 24 year olds. (Flesch Kincaid Reading Ease: 19.6, Flesch Kincaid Grade Level: 17.2, Gunning Fog Score 19.7., Coleman Liau Index: 19.5, and Automated Readability Index: 20.3).

How about breaking it into 2 sentences? Maybe something such as:

>Increased spacing between lines, words, and letters has benefits for people who need a spacing buffer. They may override author spacing settings via user stylesheet, bookmarklet, extension, or application.

That has an average grade level of about 14. It should be easily understood by 19 to 20 year olds.

Note I changed "want" to "need" as spacing is a low vision need.

Thank you!

**Comment by michael-n-cooper** [comment]
_From @lauracarlson on April 23, 2018 14:38_

@alastc wrote:

>Given the inclusion of the paragraph spacing bullet in the SC, I think the intent and benefits should include that aspect as well. E.g.

Thanks, Alastair. [I added "and paragraphs"](
https://github.com/w3c/wcag21/commit/214e5bb7d3623e32ed21ce12261f15a6976c451a).

**Comment by michael-n-cooper** [comment]
This issue did not port properly. Check [w3c/wcag21/#783](https://github.com/w3c/wcag21/issues/783) for the remainder of the context.

**Comment by patrickhlauke** [comment]
@alastc I believe this issue can be closed?

**Comment by alastc** [comment]
@patrickhlauke - agreed, I'm not seeing anything to update, I think it's out-dated.
